components/l10n/common/locale_util.h

```diff
@@ -10,13 +10,41 @@
 
 namespace brave_l10n {
 
-std::string GetLanguageCode(
-    const std::string& locale);
-
-std::string GetCountryCode(
-    const std::string& locale);
-
-std::u16string GetLocalizedResourceUTF16String(int resource_id);
+// Set the default locale for testing, restoring the original locale upon
+// destruction.
+class ScopedDefaultLocaleForTesting final {
```

Comment:
the right thing would be to put this into a separate file and include it into `testonly=true` target.
but I'm not insisting here.

Reply:
I added to locale_util because both locale_util and ScopedLocaleForTesting need access to `DefaultLocaleForTesting`